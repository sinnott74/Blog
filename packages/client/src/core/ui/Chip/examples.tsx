import React from "react";
import { Chip } from "./Chip";

export default {
    component: Chip,
    title: "Chip",
};

export const Default = () => <Chip>Test</Chip>;

export const Closeable = () => <Chip closeable>Test</Chip>;
