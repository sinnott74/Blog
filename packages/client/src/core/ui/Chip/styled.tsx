import styled from "styled-components/macro";
import { Cancel } from "../Icons";
import { Text } from "../Typography/Typography";

export const Container = styled.button`
    background: ${(props) => props.theme.palette.background.main};
    border: 0;
    border-radius: 1rem;
    padding: ${(props) => props.theme.spacing / 2 + "px"};
    transition-duration: 0.15s;
    transition-property: box-shadow, background;
    outline-style: none;
    &:hover {
        background: ${(props) => props.theme.palette.text.lighter};
    }
    &:active {
        box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 1px 5px 0 rgba(0, 0, 0, 0.12),
            0 3px 1px -2px rgba(0, 0, 0, 0.2);
    }
`;

export const Label = styled(Text)`
    display: inline-block;
    color: ${(props) => props.theme.palette.text.main};
    transition-duration: 0.15s;
    transition-property: color;
    min-height: 1.5rem;
    line-height: 1.5rem;
    padding: ${(props) => `0 ${props.theme.spacing}px`};
    margin: 0;
    ${Container}:hover & {
        color: ${(props) => props.theme.palette.background.lighter};
    }
`;

export const Close = styled(Cancel)`
    display: block;
    cursor: pointer;
    transition-duration: 0.15s;
    transition-property: fill;
    fill: ${(props) => props.theme.palette.text.main};
    ${Container}:hover & {
        fill: ${(props) => props.theme.palette.background.lighter};
    }
`;
