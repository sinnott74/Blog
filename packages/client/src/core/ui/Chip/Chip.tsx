import React, { FC } from "react";
import { Inline } from "../Layout";
import { Container, Label, Close } from "./styled";

interface ChipProps {
    children?: String;
    closeable?: boolean;
    onClick?: () => void;
}

export const Chip: FC<ChipProps> = ({ children, closeable, onClick }) => (
    <Container data-testid="chip" onClick={onClick}>
        <Inline spacing="none">
            <Label>{children}</Label>
            {closeable && <Close data-testid="chip.close-btn" />}
        </Inline>
    </Container>
);
