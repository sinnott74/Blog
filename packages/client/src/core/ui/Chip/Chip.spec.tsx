import React from "react";
import { render, fireEvent } from "../../test/test-util";
import { Chip } from "./Chip";

describe("Chip", () => {
    it("renders without crashing", () => {
        render(<Chip />);
    });

    it("matches the snapshot", () => {
        const { container } = render(<Chip>Test</Chip>);
        expect(container.firstChild).toMatchSnapshot();
    });

    it("is closable", () => {
        const wrapper = render(<Chip closeable />);
        const closeButton = wrapper.getByTestId("chip.close-btn");
        expect(closeButton).toBeTruthy();
    });

    it("calls onClick", () => {
        const handleClick = jest.fn();
        const wrapper = render(<Chip onClick={handleClick} />);
        wrapper.getByTestId("chip").click();
        expect(handleClick).toBeCalled();
    });
});
