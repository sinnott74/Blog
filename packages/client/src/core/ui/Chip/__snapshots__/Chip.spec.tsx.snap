// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip matches the snapshot 1`] = `
<button
  class="styled__Container-q6174j-0 hwjSxM"
  data-testid="chip"
>
  <p
    class="styled__Text-sc-1aodpf0-6 styled__Label-q6174j-1 ewUjkk hMQcoL"
  >
    Test
  </p>
</button>
`;
